cmd/ib-sriov-cni/main.go

```diff
@@ -215,59 +248,65 @@ func runIPAMPlugin(stdinData []byte, netConf *localtypes.NetConf) (_ *current.Re
 	return newResult, nil
 }
 
-func cmdAdd(args *skel.CmdArgs) (retErr error) {
-	netConf, netns, err := getNetConfNetns(args)
+// handleVfioPciMode handles VFIO PCI devices with VF GUID setting if needed
+func handleVfioPciMode(args *skel.CmdArgs, netConf *localtypes.NetConf, netns ns.NetNS) error {
+	// Check if the device is a VF (Virtual Function) or PF (Physical Function)
+	isVF, err := utils.IsVirtualFunction(netConf.DeviceID)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to determine if device %s is VF or PF: %v", netConf.DeviceID, err)
 	}
-	defer netns.Close()
 
-	sm := sriov.NewSriovManager()
+	// Only handle GUID setting for VF devices, keep previous code path for PF devices
+	if isVF && netConf.GUID != "" {
+		// Load device info needed for GUID setting (VFIO VF version - no network interface)
+		err = config.LoadDeviceInfoVfioVF(netConf)
```

Comment:
any reason why this should not be rolled into `configLoadDeviceInfo` which will handle both regular and vfio cases ?